deps/rabbitmq_auth_backend_oauth2/src/uaa_jwt.erl

```diff
@@ -51,6 +51,23 @@ update_uaa_jwt_signing_keys(UaaEnv0, SigningKeys) ->
     UaaEnv2 = [{signing_keys, SigningKeys} | UaaEnv1],
     application:set_env(?APP, key_config, UaaEnv2).
 
+-spec update_jwks_signing_keys() -> ok | {error, term()}.
+update_jwks_signing_keys() ->
+    UaaEnv = application:get_env(?APP, key_config, []),
+    case proplists:get_value(jwks_url, UaaEnv) of
+        undefined ->
+            {error, no_jwks_url};
+        JwksUrl ->
+            case httpc:request(JwksUrl) of
+                {ok, {_, _, JwksBody}} ->
+                    KeyList = maps:get(<<"keys">>, jose:decode(erlang:iolist_to_binary(JwksBody)), []),
+                    Keys = maps:from_list(lists:map(fun(Key) -> {maps:get(<<"kid">>, Key, undefined), {json, Key}} end, KeyList)),
+                    update_uaa_jwt_signing_keys(UaaEnv, Keys);
```

Comment:
Note: This silently overwrites `signing_keys` if the administrator specified both that and `jwks_url`.. Is there a good place to warn on startup in that case?

We can't just check whether `signing_keys` is already set, since that would break refreshing.

Reply:
Plugins can define boot steps and they can perform configuration validation. Here's one basic boot step (that has nothing to do with validation): https://github.com/rabbitmq/rabbitmq-server/blob/master/deps/rabbit/src/rabbit.erl#L251. We can add it on your behalf later.